src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1403,7 +1402,7 @@ public int getPort() {
         }
 
         @Override
-        public int getReceiveBufferSize() throws SocketException {
+        public synchronized int getReceiveBufferSize() throws SocketException {
```

Comment:
Need to replace `synchronized` with Reentrant locks, or is this case an exception?

Reply:
this is an override of Serializable